File: DesafioAlpha/app/src/main/java/com/example/desafioalpha/MainActivity.java
package com.example.desafioalpha;

import android.app.ActionBar;
import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.net.Uri;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RequestQueue requestQueue;
    private ListView lista;
    private List<Hotels> hotelsLista;
    private List<Hotels> hotelsListaOrdenada;

    //Lista para guardar os valores recebidos do JSON e popular o Listview
    private List<Hotels> Hotels;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Mostrar a Action bar e o título desta activity
        getSupportActionBar().show();

        //Titulo da Activity
        getSupportActionBar().setTitle((android.text.Html.fromHtml("<font color=\"WHITE\">"
                + "HOTÉIS" + "</font>")));


        //Popular a ListView com os dados do JSON montado acima
        lista = (ListView) findViewById(R.id.lista);

        requestQueue = Volley.newRequestQueue(this);
        Dados dados = new Dados();

        dados.ctx = this;
        dados.lista = lista;
        dados.jsonParse();

    }

}

File: HurpApp/app/src/main/java/app/recrutamento/android/challengealpha/ui/activities/MainActivity.kt
package app.recrutamento.android.challengealpha.ui.activities

import android.app.Dialog
import android.content.DialogInterface
import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import app.recrutamento.android.challengealpha.R
import app.recrutamento.android.challengealpha.model.AmenityIcon
import app.recrutamento.android.challengealpha.ui.hotellist.HotelListFragment
import app.recrutamento.android.challengealpha.util.navigateToFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(){

    lateinit var toolbar: Toolbar
    var isGridMode = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar as Toolbar?)
        txtSubtitle.setOnClickListener { withItems() }
        img_grid_on_off.setOnClickListener{ changeToListOrGrid() }
        navigateToFragment(R.id.content_frame, HotelListFragment.newInstance(), "MainActivity")
    }

    fun changeToListOrGrid() {
        var bundle = Bundle()
        if(!isGridMode){
            img_grid_on_off.setImageDrawable(getDrawable(R.drawable.ic_grid_on))
            isGridMode = true
        }else {
            img_grid_on_off.setImageDrawable(getDrawable(R.drawable.ic_grid_off))
            isGridMode = false
        }
        bundle.putBoolean("isGridMode",isGridMode)
        navigateToFragment(R.id.content_frame, HotelListFragment.newInstance(), "MainActivity",bundle)
    }


    fun withItems() {

        val dialog = Dialog(this,R.style.Theme_AppCompat_Light_Dialog_Alert)
        dialog.setContentView(R.layout.subtitle_item)
        dialog.setTitle("Lista de Legendas")
        dialog.show()
        
    }

}
